_drafts/produccion-y-desarollo-en-django.md

```diff
@@ -0,0 +1,85 @@
+---
+title: "Produccion y desarollo en Django"
+date: 2019-03-09
+author: Esteban Galicia
+tags: Django producción deploy despliegue
+categories: 
+comments: true
+excerpt: "Una manera para manejar producción y desarrollo sin tanto esfuerzo."
+header:
+  overlay_image: https://res.cloudinary.com/dg89awi0p/image/upload/v1552160611/rawpixel-1135756-unsplash.jpg
+  teaser: https://res.cloudinary.com/dg89awi0p/image/upload/v1552160611/rawpixel-1135756-unsplash.jpg
+---
+
+
+# Para no salir herido al combinar desarrollo y producción en Django.
+
+
+> Se usará Django >= 2.0
+
+> Se usará Python >= 3.6
+
+## Problemática.
+
+Al desarrollar en Django se suscitó un problema, se requería hacer múltiples pruebas con el ORM para generar consultas complejas a base de datos.
+Para hacer la prueba se ejecuta:
+```python
+python manage.py shell
+```
+Con lo que se abrirá una consola de Python en dicha consola se puede importar un modelo y con ese modelo hacer las pruebas requeridas.
+
+```python
+from blog.models import Post
+
+all_posts = post.objects.all()
+```
+
+Todo opera bien hasta que debemos importar múltiples modelos, al modificar un modelo se debe detener la consola `Ctrl+C` y ejecutarla de nueva cuenta, y de nueva cuenta cargar todos los modelos.
+
+A fin de perder tiempo cargando los modelos en cada detención de la consola se opta por usar una herramienta llamada [`django-extensions`](https://django-extensions.readthedocs.io/en/latest/)
+
+La instalo en mi proyecto usando
+```sh
+pip install 
+```
+
+y agregándola a mis apps
+
+```python
+INSTALLED_APPS = (
+    ...
+    'django_extensions',
+)
+```
+
+Hecho eso, puedo ejecutar una consola donde se cargan ya todos los modelos listos para hacer consultas a la base de datos, traducción, ahorramos mucho tiempo.
```

Comment:
...consultas a la base de datos. Traduccióin: ahorramos mucho tiempo. <- Creo que eso podría mejorarse así.

Reply:
@hectorip listo